### ISSUE TITLE
Optimization: deconstruction of tuple literal should not construct a ValueTuple

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#13631]
From LDM discussion today, there was a desire to optimize such case and use temps instead (to respect the evaluation order), or even directly use the locals when possible.

``` C#
        [Fact]
        public void DeconstructionDeclaration()
        {
            string source = @"
class C
{
    static void Main()
    {
        var (x1, x2) = (1, ""hello"");
        System.Console.WriteLine(x1 + "" "" + x2);
    }
}
";

            var comp = CompileAndVerify(source, additionalRefs: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, expectedOutput: "1 hello");
            comp.VerifyDiagnostics();
            comp.VerifyIL("C.Main", @"
{
  // Code size       49 (0x31)
  .maxstack  3
  .locals init (int V_0, //x1
                string V_1, //x2
                int V_2)
  IL_0000:  ldc.i4.1
  IL_0001:  ldstr      ""hello""
  IL_0006:  newobj     ""System.ValueTuple<int, string>..ctor(int, string)""
  IL_000b:  dup
  IL_000c:  ldfld      ""int System.ValueTuple<int, string>.Item1""
  IL_0011:  stloc.2
  IL_0012:  ldfld      ""string System.ValueTuple<int, string>.Item2""
  IL_0017:  ldloc.2
  IL_0018:  stloc.0
  IL_0019:  stloc.1
  IL_001a:  ldloc.0
  IL_001b:  box        ""int""
  IL_0020:  ldstr      "" ""
  IL_0025:  ldloc.1
  IL_0026:  call       ""string string.Concat(object, object, object)""
  IL_002b:  call       ""void System.Console.WriteLine(string)""
  IL_0030:  ret
}
");
        }
```

**Comment by gafter** [comment]
This will probably be easier to address in lowering when the phase ordering error described in #13270 has been addressed.

**Comment by jcouv** [comment]
Fixed by #17027
